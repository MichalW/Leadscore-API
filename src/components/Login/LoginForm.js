import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = (event) => {
    this.setState({username: event.target.value});
  };

  onChangePassword = (event) => {
    this.setState({password: event.target.value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          <Button type="submit" fluid primary>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
