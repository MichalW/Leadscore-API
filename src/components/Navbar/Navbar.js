import PropTypes from 'prop-types';
import React from 'react';
import {Container, Menu} from 'semantic-ui-react';

const Navbar = ({signOutRequest}) => (
  <Menu size="huge" fixed="top">
    <Container>
      <Menu.Item name="home" onClick={() => {}} />

      <Menu.Menu position="right">
        <Menu.Item name="logout" onClick={signOutRequest} />
      </Menu.Menu>
    </Container>
  </Menu>
);

Navbar.propTypes = {
  signOutRequest: PropTypes.func.isRequired,
};

export default Navbar;
