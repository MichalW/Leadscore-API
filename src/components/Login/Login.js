import React from 'react';
import PropTypes from 'prop-types';
import {Grid, Header} from 'semantic-ui-react';

import LoginForm from './LoginForm';

const Login = ({signInRequest}) => (
  <Grid columns={3} centered verticalAlign="middle" style={{minHeight: '100vh'}} className="background-blue">
    <Grid.Row>
      <Grid.Column>
        <Header as="h1">
          <strong>Lead</strong>
          score
        </Header>

        <LoginForm signInRequest={signInRequest} />
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

Login.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default Login;
