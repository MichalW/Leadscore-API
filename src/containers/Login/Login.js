import React from 'react';
import {Grid, Header} from 'semantic-ui-react';

import LoginForm from '../../components/LoginForm';
import styles from './Login.module.scss';

const Login = () => (
  <Grid
    centered
    className={styles.wrapper}
    columns={3}
    style={{minHeight: '100vh'}}
    verticalAlign="middle"
  >
    <Grid.Row>
      <Grid.Column>
        <Header as="h1">
          <strong>Lead</strong>
          score
        </Header>

        <LoginForm />
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

export default Login;
