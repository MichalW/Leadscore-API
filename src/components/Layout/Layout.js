import PropTypes from 'prop-types';
import React from 'react';
import {Container, Grid} from 'semantic-ui-react';

import Navbar from '../Navbar';

const Layout = ({children}) => (
  <Container>
    <Navbar />

    <Grid centered>
      <Grid.Row>
        <Grid.Column>
          <Container>
            {children}
          </Container>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
