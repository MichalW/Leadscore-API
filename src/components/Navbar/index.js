import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Navbar from './Navbar';
import {signOutRequest} from '../../actions/authentication';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
});

const mapDispatchToProps = dispatch => ({
  signOutRequest: bindActionCreators(signOutRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
