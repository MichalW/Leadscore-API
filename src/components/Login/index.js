import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Login from './Login';
import {signInRequest} from '../../actions/authentication';

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(null, mapDispatchToProps)(Login);
