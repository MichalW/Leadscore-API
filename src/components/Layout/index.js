import {connect} from 'react-redux';

import Layout from './Layout';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Layout);
