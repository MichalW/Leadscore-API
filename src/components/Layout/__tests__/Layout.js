import React from 'react';
import {shallow} from 'enzyme';

import Layout from '../Layout';
import Navbar from '../../Navbar';
import Login from '../../Login';

it('renders Layout signedIn', () => {
  const props = {
    signedIn: true,
    username: 'ExampleUsername',
  };

  const layout = shallow(<Layout {...props} />);

  expect(layout.find(Navbar).exists()).toBe(true);
  expect(layout.find(Login).exists()).toBe(false);
});

it('renders Layout signedOut', () => {
  const props = {
    signedIn: false,
  };

  const layout = shallow(<Layout {...props} />);

  expect(layout.find(Login).exists()).toBe(true);
});
