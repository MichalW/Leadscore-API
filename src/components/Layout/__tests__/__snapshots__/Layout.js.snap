// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout 1`] = `
<Container>
  <Connect(Navbar) />
  <Grid
    centered={true}
  >
    <GridRow>
      <GridColumn>
        <Container>
          <ExamplePage />
        </Container>
      </GridColumn>
    </GridRow>
  </Grid>
</Container>
`;
